### PR TITLE
integration of registeration api 

### LINKED ISSUE #17: Integrate the Registration API
Track the progress of [this](https://github.com/SPO-Career-Portal/SCP-Backend/pull/26) PR. Connect it using preferably fetch API or react-axios.

### PR DESCRIPTION
**Pull request by navneet-28**
Integration of Registration API and made required changes in the registration process in the frontend in accordance with the backend.
resolves #17

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -14,21 +29,36 @@ export const login = (username) => async () => {
...
 
+export const setPass = (password) => async ()  => {
+  try{
+    const pass=password;
+    const result= sendRequest("/user/register/verify/code=<str:token>/", "POST", {
+      pass,
+    });
+    if(result=="200_OK"){
+      sessionService.saveSession();
+      sessionService.saveUser(pass);
```
Comment (mridul911) [inline src/actions/userActions.js]: Don't save passwords in session. If needed save some other details.

File: src/views/examples/Register.js [hunk 2]
```diff
@@ -24,145 +25,83 @@ import {
...
-    if(emailError==="valid email"&& password==confirmPassword){
-      if (e) e.preventDefault();
+
+  const [roll, setRoll] = useState("");
+  const history = useHistory();
+
+
+  const onSubmit = async (e)=> {
+    const id=1234;
+    history.push("/auth/setPass/access/:id")
```
Comment (mridul911) [inline src/views/examples/Register.js]: you can remove this now and check by changing URL in backend for mail it should match prefix of current URL

File: src/views/examples/Register.js [hunk 3]
```diff
@@ -24,145 +25,83 @@ import {
...
                   </InputGroupAddon>
                   <Input required valid
-                    class="is-invalid from-control"
-                    onChange={(e) => confirmpassword(e.target)}
-                    placeholder="Confirm Your Password"
-                    type="password"
+                    placeholder="IITK Roll no."
+                    type="roll"
                     autoComplete="off"
+                    onChange ={(e) => setRoll(e.target.value)}
```
Comment (mridul911) [inline src/views/examples/Register.js:84]: Make a popup if that roll number has been used. Something like an error: roll number is in use try to log in.

**Review by mridul911 — CHANGES_REQUESTED** [review]
Please make changes in setting_email.py in your local system and check whether we are able to set a password for it.

**Comment by navneet-28** [comment]
applied all changes except the "roll no. already in use" one.

**Comment by mridul911** [comment]
> applied all changes except the "roll no. already in use" one.

for this you need to check the response from backend if it is forbidden then may be user is already registered.Kindly make that change else looks everything look fine. Great job!

**Review by mridul911 — APPROVED** [review]

**Pull request merged**